xapian-core/common/parseint.h

```diff
@@ -0,0 +1,40 @@
+/** @file parseint.h
+ * @brief Parse unsigned type from string and check for trailing characters.
+ */
+/* Copyright (C) 2018 Olly Betts
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+#ifndef XAPIAN_INCLUDED_PARSEINT_H
+#define XAPIAN_INCLUDED_PARSEINT_H
+
+#include "overflow.h"
+
+template<typename T>
+bool parse_unsigned(const char* p, T& res)
+{
+    res = 0;
+    do {
+	unsigned char digit = *p - '0';
+	if (digit > 9 ||
+	    mul_overflows(res, (unsigned int)10, res) ||
+	    add_overflows(res, digit, res)) {
+	return false;
+    }
+} while (*++p);
+return true;
```

Comment:
The four lines above are each indented one level too few.

Reply:
will fix it